### ISSUE TITLE
Update icons that are referenced by unicode characters

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#116136]
When using product icon themes, icons that are referenced via unicode characters aren't replaced. We should update the code to have these as added as theme icons. Below is the search query I did in the codebase:

Query: `content: "\`

```
11 results - 9 files

src/vs/base/browser/ui/menu/menubar.css:
  85  .menubar.compact .toolbar-toggle-more::before {
  86: 	content: "\eb94" !important;
  87  }

src/vs/workbench/browser/actions/media/actions.css:
  6  .monaco-workbench .quick-input-list .quick-input-list-entry.has-actions:hover .quick-input-list-entry-action-bar .action-label.dirty-workspace::before {
  7: 	content: "\ea76"; /* Close icon flips between black dot and "X" for dirty workspaces */
  8  }

src/vs/workbench/browser/media/style.css:
  192  .monaco-workbench .select-container:after {
  193: 	content: "\eab4";
  194  	font-family: codicon;

src/vs/workbench/browser/parts/editor/media/editorquickaccess.css:
  6  .quick-input-list .quick-input-list-entry.has-actions:hover .quick-input-list-entry-action-bar .action-label.dirty-editor::before {
  7: 	content: "\ea76"; /* Close icon flips between black dot and "X" for dirty open editors */
  8  }

src/vs/workbench/browser/parts/editor/media/tabstitlecontrol.css:
  322  .monaco-workbench .part.editor > .content .editor-group-container > .title .tabs-container > .tab.sticky.dirty > .tab-actions .action-label:not(:hover)::before {
  323: 	content: "\ebb2"; /* use `pinned-dirty` icon unicode for sticky-dirty indication */
  324  }

  327  .monaco-workbench .part.editor > .content .editor-group-container > .title .tabs-container > .tab.dirty > .tab-actions .action-label:not(:hover)::before {
  328: 	content: "\ea71"; /* use `circle-filled` icon unicode for dirty indication */
  329  }

src/vs/workbench/contrib/files/browser/views/media/openeditors.css:
  30  .open-editors .monaco-list .monaco-list-row.dirty:not(:hover) > .monaco-action-bar .codicon-pinned::before {
  31: 	content: "\ebb2"; /* use `pinned-dirty` icon unicode for sticky-dirty indication */
  32  }

  34  .open-editors .monaco-list .monaco-list-row.dirty:not(:hover) > .monaco-action-bar .codicon-close::before {
  35: 	content: "\ea71"; /* use `circle-filled` icon unicode for dirty indication */
  36  }

src/vs/workbench/contrib/search/browser/media/anythingQuickAccess.css:
  6  .quick-input-list .quick-input-list-entry.has-actions:hover .quick-input-list-entry-action-bar .action-label.dirty-anything::before {
  7: 	content: "\ea76"; /* Close icon flips between black dot and "X" for dirty open editors */
  8  }

src/vs/workbench/electron-sandbox/actions/media/actions.css:
  6  .monaco-workbench .quick-input-list .quick-input-list-entry.has-actions:hover .quick-input-list-entry-action-bar .action-label.dirty-window::before {
  7: 	content: "\ea76"; /* Close icon flips between black dot and "X" for dirty windows */
  8  }

src/vs/workbench/services/decorations/browser/decorationsService.ts:
  91  			`.${this.bubbleBadgeClassName}::after`,
  92: 			`content: "\uea71"; color: ${getColor(theme, color)}; font-family: codicon; font-size: 14px; padding-right: 14px; opacity: 0.4;`,
  93  			element

```

**Comment by misolori** [comment]
Ran into this issue with my product icon theme where adding a close icon, the dirty indicator no longer showed up https://github.com/misolori/vscode-fluent-icons/issues/3

**Comment by misolori** [comment]
The dirty/pin indicators seem to be the biggest offenders here. The code pointer is here:

https://github.com/microsoft/vscode/blob/de4e9067e80f079f092006929dab05516e27c70e/src/vs/workbench/browser/parts/editor/tabsTitleControl.ts#L1240-L1242

I attempted to fix this but ultimately needs a better fix that can listen for mouse hovers to swap between the close/dot/pin icons using the proper `ThemeIcon` constructor. @bpasero any idea on the best way to achieve this since this is in your area?

**Comment by bpasero** [comment]
Since I don't really understand what needs to happen here, it would help if `I attempted to fix this...` points to a commit of your attempt.

**Comment by misolori** [comment]
TL;DR we need to move away from referencing unicode characters directly as they are not themeable. 

So these are changes that need to happen for using which icons:

## Dirty state
- When a file is dirty  => `codicon-circle-filled`
- When a file is dirty and hovered => `codicon-close`

## Pinned state
- When a file is pinned => `codicon-pinned`
- When a file is pinned and dirty => `codicon-pinned-dirty`
- When a file is pinned and dirty and hovered  => `codicon-pinned`

[I have a branch](https://github.com/microsoft/vscode/tree/misolori/dirty-tabs) that attempted to do this but no very well. Even then, this wouldn't scale well because I need to swap the icons on hover as well.

**Comment by bpasero** [comment]
Thanks for clarifying, I think doing it through code is the only way and your change seems like a good start for that. But I definitely see the challenge with trying to replicate complex CSS rules. We want the icon to change on hover to reflect the action that happens when you click (i.e. the circle changes to the close icon). 

Is there no way for us to reference codicons through CSS while presering the theming? Wondering if another approach is possible, maybe via CSS variables?
